### PR TITLE
fix toFile call

### LINKED ISSUE #1297: Remove Path.toFile in IntervalListWriter
#1288 has a call of Path.toFile which should be removed and replaced with a proper path call but is blocked by #1296

### PR DESCRIPTION
**Pull request by lbergelson**
### Description

fixes #1297 and does some nitpicky formatting, blocked by #1296 
### Checklist

- [ ] Code compiles correctly
- [ ] New tests covering changes and new functionality
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

File: src/test/java/htsjdk/samtools/util/IntervalCodecTest.java [hunk 1]
```diff
@@ -59,27 +59,26 @@ public void testEndToEnd() throws IOException  {
...
-        final InputStream inputStream = new FileInputStream(tempFile);
-        readCodec.setInputStream(inputStream);
-        while (true) {
-            final Interval interval = readCodec.decode();
-            if (interval == null) {
-                break;
-            }
-            else {
-                actualList.add(interval);
+        try(final InputStream inputStream = new FileInputStream(tempFile)) {
```
Comment (pshapiro4broad) [inline src/test/java/htsjdk/samtools/util/IntervalCodecTest.java]:
```suggestion
        try (final InputStream inputStream = new FileInputStream(tempFile)) {
```

File: src/test/java/htsjdk/samtools/util/IntervalListWriterTest.java [hunk 2]
```diff
@@ -26,7 +32,18 @@ void setup() {
     public void testEndToEnd() throws IOException {
         final File tempFile = File.createTempFile("IntervalListWriterTest.", ".interval_list");
         tempFile.deleteOnExit();
+        testEndToEnd(tempFile.toPath());
+    }
+
+    @Test
+    public void testEndToEndOnPath() throws IOException {
+        try(FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
```
Comment (pshapiro4broad) [inline src/test/java/htsjdk/samtools/util/IntervalListWriterTest.java]:
```suggestion
        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
```

File: src/main/java/htsjdk/samtools/util/IntervalListWriter.java [hunk 3]
```diff
@@ -67,23 +67,19 @@ public IntervalListWriter(final  Path path, final SAMFileHeader header) {
...
         out.write(interval.getContig());
-        out.write('\t');
+        out.write(TAB);
         out.write(Integer.toString(interval.getStart()));
-        out.write('\t');
+        out.write(TAB);
         out.write(Integer.toString(interval.getEnd()));
-        out.write('\t');
+        out.write(TAB);
         out.write(interval.isPositiveStrand() ? '+' : '-');
```
Comment (yfarjoun) [inline src/main/java/htsjdk/samtools/util/IntervalListWriter.java]: Perhaps use `Strand.POSITIVE` and `Strand.NEGATIVE` while you're here?

File: src/test/java/htsjdk/samtools/util/IntervalCodecTest.java [hunk 4]
```diff
@@ -59,27 +59,26 @@ public void testEndToEnd() throws IOException  {
         expectedList.add(new Interval("chr1", 50, 150, true, "number-5"));
         expectedList.add(new Interval("chr1", 150, 250, false, "number-6"));
 
-        final OutputStream outputStream = new FileOutputStream(tempFile);
-        final IntervalCodec writeCodec = new IntervalCodec(this.dict);
-        writeCodec.setOutputStream(outputStream);
-        for (final Interval interval : expectedList.getIntervals()) {
-            writeCodec.encode(interval);
+        try(final OutputStream outputStream = new FileOutputStream(tempFile)) {
+            final IntervalCodec writeCodec = new IntervalCodec(this.dict);
```
Comment (yfarjoun) [inline src/test/java/htsjdk/samtools/util/IntervalCodecTest.java]: need `this`?

File: src/test/java/htsjdk/samtools/util/IntervalCodecTest.java [hunk 5]
```diff
@@ -59,27 +59,26 @@ public void testEndToEnd() throws IOException  {
...
+            writeCodec.setOutputStream(outputStream);
+            for (final Interval interval : expectedList.getIntervals()) {
+                writeCodec.encode(interval);
+            }
         }
-        outputStream.close();
 
         final IntervalCodec readCodec = new IntervalCodec(this.dict);
```
Comment (yfarjoun) [inline src/test/java/htsjdk/samtools/util/IntervalCodecTest.java]: need 'this'?

File: src/test/java/htsjdk/samtools/util/IntervalListWriterTest.java (hunk 2 above)
Comment (yfarjoun) [inline src/test/java/htsjdk/samtools/util/IntervalListWriterTest.java]: what does `Configuration.unix()` do for us? can you dd a comment to explain this test?

File: src/main/java/htsjdk/samtools/util/IntervalListWriter.java (hunk 3 above)
Comment (lbergelson) [reply src/main/java/htsjdk/samtools/util/IntervalListWriter.java]: Done

File: src/test/java/htsjdk/samtools/util/IntervalCodecTest.java (hunk 4 above)
Comment (lbergelson) [reply src/test/java/htsjdk/samtools/util/IntervalCodecTest.java]: nope.  I removed it from the rest of the class too except where needed.

File: src/test/java/htsjdk/samtools/util/IntervalCodecTest.java (hunk 5 above)
Comment (lbergelson) [reply src/test/java/htsjdk/samtools/util/IntervalCodecTest.java]: gone

File: src/test/java/htsjdk/samtools/util/IntervalCodecTest.java [hunk 6]
```diff
@@ -59,27 +59,26 @@ public void testEndToEnd() throws IOException  {
...
-            }
-            else {
-                actualList.add(interval);
+        try(final InputStream inputStream = new FileInputStream(tempFile)) {
+            readCodec.setInputStream(inputStream);
+            while (true) {
+                final Interval interval = readCodec.decode();
+                if (interval == null) {
+                    break;
+                } else {
```
[the remark this answers is not in the document]
Comment (lbergelson) [reply src/test/java/htsjdk/samtools/util/IntervalCodecTest.java]: It's unnecessary but I don't mind them.  I always think it makes it more clear from just looking at the structure of the code what the paths are.  I've removed it here.

File: src/test/java/htsjdk/samtools/util/IntervalListWriterTest.java (hunk 2 above)
Comment (lbergelson) [reply src/test/java/htsjdk/samtools/util/IntervalListWriterTest.java]: It makes the file system behave like a unix one vs a windows/ mac one.   Added a comment.

File: src/main/java/htsjdk/samtools/util/IntervalListWriter.java (hunk 3 above)
Comment (lbergelson) [reply src/main/java/htsjdk/samtools/util/IntervalListWriter.java]: I added a new method Interval.getStrand() that returns a Strand object that knows its string representation.

**Comment by lbergelson** [comment]
@yfarjoun Any further comments on this?  I added a new simple method to Interval to return the `Strand` enum.

File: src/test/java/htsjdk/samtools/util/IntervalTest.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    @Test(dataProvider = "booleanProvider")
+    public void testGetStrand(boolean isNegativeStrand){
+        final Interval interval = new Interval("chr1", 1, 100, isNegativeStrand, "name");
+        Assert.assertEquals(isNegativeStrand, interval.isNegativeStrand());
+        Assert.assertNotEquals(isNegativeStrand, interval.isPositiveStrand());
+        Assert.assertEquals(isNegativeStrand, interval.getStrand() == Strand.NEGATIVE);
+    }
+}
```
Comment (yfarjoun) [inline src/test/java/htsjdk/samtools/util/IntervalTest.java]: newline

**Pull request merged**